src/prefect/engine.py

```diff
@@ -254,27 +253,34 @@ async def retrieve_flow_then_begin_flow_run(
         )
         return state
 
+    parameters = flow_run.parameters
+
     await client.update_flow_run(
         flow_run_id=flow_run_id,
         flow_version=flow.version,
+        parameters=parameters,
```

Comment:
I think the reason we were doing the prior code was to retrieve default values for unprovided parameters. No reason to update the flow run parameters in this following call if we're not mutating them.